google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -0,0 +1,347 @@
+  }
+
+  /**
+   * Tests if a file is valid by checking whether it exists, is a regular file and is well formed.
+   */
+  private boolean isValidConfigurationFile(String path) {
+    try {
+      if (!Files.exists(Paths.get(path))
+          || !Files.isRegularFile(Paths.get(path))) {
+        return false;
+      }
+    } catch (InvalidPathException ipe) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Validates the configuration and turns on/off any necessary warnings.
+   */
+  private void validateConfiguration() {
+    boolean showError = false;
+
+    if (!isValidConfigurationFile(yaml.getText())) {
+      errorIcon.setVisible(true);
+      errorMessage.setVisible(true);
+      errorMessage.setText(GctBundle.getString("appengine.deployment.error.staging.yaml"));
+      showError = true;
+    }
+
+    if (isRuntimeCustom() && !isValidConfigurationFile(dockerfile.getText())) {
+      errorIcon.setVisible(true);
+      errorMessage.setVisible(true);
+      errorMessage.setText(GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+      showError = true;
+    }
+
+    errorIcon.setVisible(showError);
+    errorMessage.setVisible(showError);
+  }
+
+  private void toggleDockerfileSection() {
+    boolean enabled = isRuntimeCustom();
+    dockerfile.setEnabled(enabled);
+    genDockerfileButton.setEnabled(enabled);
+    // Shows no Dockerfile label if runtime is Java and yaml is valid.
+    noDockerfileLabel.setVisible(!enabled && isValidConfigurationFile(yaml.getText()));
+  }
+
+  @Override
+  public void onFacetInitialized(@NotNull Facet facet) {
+    if (facet instanceof AppEngineFlexibleFacet) {
+      ((AppEngineFlexibleFacet) facet).getConfiguration().setYamlPath(yaml.getText());
+      ((AppEngineFlexibleFacet) facet).getConfiguration().setDockerfilePath(dockerfile.getText());
+    }
+  }
+
+  /**
+   * A somewhat generic way of generating a file for a {@link TextFieldWithBrowseButton}.
+   */
+  private static class GenerateConfigActionListener implements ActionListener {
+
+    private final Project project;
+    private final String fileName;
+    private final TextFieldWithBrowseButton filePicker;
+    private final Supplier<Optional<Path>> sourceFileProvider;
+    // Used to refresh the warnings.
+    private final Runnable callback;
+
+    GenerateConfigActionListener(
+        Project project,
+        String fileName,
+        Supplier<Optional<Path>> sourceFileProvider,
+        TextFieldWithBrowseButton filePicker,
+        Runnable callback) {
+      this.project = project;
+      this.fileName = fileName;
+      this.sourceFileProvider = sourceFileProvider;
+      this.filePicker = filePicker;
+      this.callback = callback;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent event) {
+      Path sourceFile = sourceFileProvider.get().orElseThrow(
+          () -> new AssertionError("appengine.deployment.error.deployable.notjarorwar"));
```

Comment:
Ah, this made sense when Dockerfile generation depended on the chose user specified artifact. Because we're no longer using this in that context (and are assuming WAR format for every Dockerfile generated in this way), it's pretty much guaranteed to always return.